ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -143,6 +145,20 @@ private void onBlockAdded(final BlockAddedEvent blockAddedEvent) {
                   LOG.info("Imported {} pending blocks", r.size());
                 }
               });
+    } else {
+      pendingBlocksManager
+          .lowestAnnouncedBlock()
+          .map(ProcessableBlockHeader::getNumber)
+          .ifPresent(
+              minAnnouncedBlockNumber -> {
+                long distance =
+                    minAnnouncedBlockNumber
+                        - protocolContext.getBlockchain().getChainHeadBlockNumber();
+                if (distance < config.getBlockPropagationRange().upperEndpoint()
+                    && minAnnouncedBlockNumber > newBlock.getHeader().getNumber()) {
+                  retrieveMissingAnnouncedBlock(newBlock.getHeader().getNumber() + 1);
+                }
```

Comment:
We already check that a block is within the import range in `handleNewBlockFromNetwork`. Do you see an issue in always attempting to get the `chainHead +1`th block if there are any saved blocks? For example:

```java
      long nextExpectedBlockNumber = newBlock.getHeader().getNumber() + 1;
      if (pendingBlocksManager
          .lowestAnnouncedBlock()
          .map(ProcessableBlockHeader::getNumber)
          .filter(minAnnouncedBlockNumber -> minAnnouncedBlockNumber > nextExpectedBlockNumber)
          .isPresent()) {
        retrieveMissingAnnouncedBlock(nextExpectedBlockNumber);
      }
```

Reply:
this is to avoid triggering this mechanism when we are starting a fullsync from scratch for example and/or when we are too far from the head. Just to use this when we are in a "block by block" kind of sync.

Reply:
Ah I see. Sounds good.